packages/blocks/src/attachment-block/components/options.ts

```diff
@@ -0,0 +1,264 @@
+import {
+  CaptionIcon,
+  DeleteIcon,
+  DownloadIcon,
+  DuplicateIcon,
+  EditIcon,
+  EmbedWebIcon,
+  LinkIcon,
+  MoreIcon,
+  ViewIcon,
+} from '@blocksuite/global/config';
+import { assertExists } from '@blocksuite/store';
+import { css, html } from 'lit';
+import { createRef, type Ref, ref } from 'lit/directives/ref.js';
+
+import { stopPropagation } from '../../__internal__/utils/event.js';
+import { getViewportElement } from '../../__internal__/utils/query.js';
+import { tooltipStyle } from '../../components/tooltip/tooltip.js';
+import type { ImageProps } from '../../image-block/image-model.js';
+import type { AttachmentBlockModel } from '../attachment-model.js';
+import { cloneAttachmentProperties, downloadAttachment } from '../utils.js';
+
+const styles = css`
+  .affine-attachment-options {
+    position: fixed;
+    left: 0;
+    top: 0;
+
+    display: flex;
+    align-items: center;
+    padding: 8px;
+    gap: 8px;
+    border-radius: 8px;
+    background: var(--affine-background-overlay-panel-color);
+    box-shadow: var(--affine-shadow-2);
+  }
+
+  .affine-attachment-options .divider {
+    width: 1px;
+    height: 24px;
+    background-color: var(--affine-border-color);
+  }
+
+  ${tooltipStyle}
+`;
+
+export function AttachmentOptionsTemplate({
+  anchor,
+  model,
+  abortController,
+}: {
+  anchor: HTMLElement;
+  model: AttachmentBlockModel;
+  abortController: AbortController;
+}) {
+  let hoverTimeout = 0;
+  const onHover = () => clearTimeout(hoverTimeout);
+  const onHoverLeave = () => {
+    const HOVER_TIMEOUT = 300;
+    clearTimeout(hoverTimeout);
+    hoverTimeout = window.setTimeout(() => {
+      abortController.abort();
+    }, HOVER_TIMEOUT);
+  };
+  anchor.addEventListener('mouseover', onHover);
+  anchor.addEventListener('mouseleave', onHoverLeave);
+  abortController.signal.addEventListener('abort', () => {
+    anchor.removeEventListener('mouseover', onHover);
+    anchor.removeEventListener('mouseleave', onHoverLeave);
+  });
+
+  const containerRef = createRef<HTMLDivElement>();
+  const updatePosition = () => {
+    const container = containerRef.value;
+    if (!container) return;
+    const { top, left, width } = anchor.getBoundingClientRect();
+    container.style.transform = `translate(calc(${
+      left + width
+    }px - 100%), calc(${top}px - 100% - 4px))`;
+  };
+
+  const viewportElement = getViewportElement(model.page);
+  if (viewportElement) {
+    viewportElement.addEventListener('scroll', updatePosition);
+  }
+
+  setTimeout(() => {
```

Comment:
Is this required for all portal component users?

Reply:
The key to `setTimeout` lies in the `containerRef`. In order to update the position of the HTMLElement, specific DOM information is required.

Reply:
Is it possible to encapsulate these logic into some util methods together with `createLitPortal`? The current API surface looks a bit too verbose and fragmented.

Reply:
We can encapsulate a higher-level API together as a higher-level API in the future.

```ts
createPopoverPortal({
  reference,
  template, // <- portal template
  positionOptions,
  ... createLitPortalOptions
})
```

Ref API
```ts
import { computePosition, offset } from '@floating-ui/dom';

  computePosition(reference, menu, {
    placement: 'top',
    middleware: [
      offset({
        mainAxis: 10,
      }),
    ],
  }).then(({ x, y }) => {
    Object.assign(menu.style, {
      left: `${x}px`,
      top: `${y}px`,
    });
  });
```

We can also add lifecycle to `createLitPortal`, but this will reduce the understanding and convenience of createLitPortal.

```ts
const createLitPortal({
  render: ({ onMount, befortUnmount }) => TemplateResult<1>;
  container?: HTMLElement;
  abortController?: AbortController;
}): HTMLDivElement
```